database/postgresql_adapter.go

```diff
@@ -0,0 +1,54 @@
+package database
+
+import (
+	"fmt"
+
+	"github.com/bitcoin-sv/pulse/config"
+	"github.com/golang-migrate/migrate/v4"
+	postgres "github.com/golang-migrate/migrate/v4/database/postgres"
+
+	// use blank import to use file source driver with the migrate package.
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/jmoiron/sqlx"
+
+	// use blank import to register postgresql driver.
+	_ "github.com/lib/pq"
+)
+
+type PostgreSqlAdapter struct{}
+
+const postgresDriverName = "postgres"
+
+func (a *PostgreSqlAdapter) Connect(cfg *config.DbConfig) (*sqlx.DB, error) {
+	dbCfg := cfg.Postgres
```

Comment:
Do we do some validations to prevent the case when user selects in config `Type` as `postgres` but doesn't set `Postgres` structure? In this case, the `dbCfg` can be `nil` what leads to nil ptr panic.

Reply:
Good point. I validate config at appstart